typegen/generate_typing.py

```diff
@@ -0,0 +1,157 @@
+from pathlib import Path
+from typing import Final
+
+
+def write_annotations(output_folder: Path, strict: bool) -> None:
+    """Generate and write typing annotations for the library
```

Comment:
We use the [NumPyDoc-format](https://numpydoc.readthedocs.io/en/latest/format.html)

I'll make this cleared in `CONTRIBUTOR.md` in a future PR, apologies.